packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2669,6 +2669,7 @@ class RenderMouseRegion extends RenderProxyBox {
       onEnter: _handleEnter,
       onHover: _handleHover,
       onExit: _handleExit,
+      getTransform: () => getTransformTo(null),
```

Comment:
Is this gonna be ok for performance?

In the touch pointer world we calculate the transform during hit testing to avoid walking the render tree multiple times (for performance reasons).
And I thought in the mouse world we decided to do hit testing on the layer tree because walking the render tree is too much. However, I believe `getTransformTo` is walking the render tree to figure out the transform. So, we've implicitly added walking the render tree to mouse hit testing?

Reply:
If I understand the code correctly it doesn't really walk the tree, but traverse the ancestors. So it doesn't really walk the entire tree, although it does perform _n_ matrix multiplications, where _n_ is the depth of the node.

Reply:
@goderbauer If I want to cache the transforming matrix, is it reliable to clear the cache during `paint`?

Reply:
Can we avoid traversing the ancestors by collecting the transformation matrix during hit testing?

I believe caching and clearing in paint doesn't work because of RepaintBoundaries. If you're a RepaintBoundary the transform may change, but you may not be told to repaint.